torchio/data/queue.py

```diff
@@ -146,15 +147,39 @@ def __init__(
         self.shuffle_subjects = shuffle_subjects
         self.shuffle_patches = shuffle_patches
         self.samples_per_volume = samples_per_volume
+        if type(samples_per_volume) == int:
+            self.samples_per_volume = [samples_per_volume for _ in range(
+                self.num_subjects)]
+        assert type(self.samples_per_volume) == list, ('`samples_per_volume`'
+                                                       'should be an int'
+                                                       'or a list')
+        assert len(self.samples_per_volume) == self.num_subjects, (
+            '`samples_per_volume` (list) length must be equal'
+            'to the number of subjects')
         self.sampler = sampler
         self.num_workers = num_workers
         self.verbose = verbose
         self._subjects_iterable = None
-        if start_background:
-            self._initialize_subjects_iterable()
         self.patches_list: List[Subject] = []
         self.num_sampled_patches = 0
 
+        # Same random shuffling applied to subjects and volumes
+        if self.shuffle_subjects:
+            tmp_seed = np.random.random()
```

Comment:
Is it possible to use PyTorch instead of NumPy for this? NumPy random operations are sometimes problematic in this context.